### PR TITLE
You can now click on objects and mobs below flaps and signs

### LINKED ISSUE #41348: Tactical atmos holobarriers
Round ID: 96742
Atmos holobarriers are layered above mobs, making it so you can't directly click them in order to melee.

Possible fixes include making the sprite smaller, or make it so it isn't layered above mobs.

Issue stems from a round where a group of players on the arrivals escape pod were able to hold out against a conventional invasion where ultimately I was unable to defeat them due being unable to click them with a stunprod due to hitting the holobarrier instead.
(alt clicking the turf would result in me getting disarmed and it isn't suitable for combat)

### LINKED ISSUE #41589: tactical plastic flaps
1. Punch someone and start a fight
2. Run and crawl under flaps
3. Have free reign to click them while they can’t click back, double points in you click with a stun weapon so they don’t even get a chance to alt click the tile and click you.

Might not seem too exploitable at first but it wouldn’t be that difficult to set up multiple flaps around places you might be fighting, or baiting people to parts of departments with them.

### PR DESCRIPTION
**Pull request by nicbn**
[Changelogs]: #
:cl:
fix: You can click on things that are under flaps or holo barriers.
/:cl:

Closes #41589
Closes #41348

Successor of #41854

### CODE REVIEW

**Comment by yoyobatty** [comment]
My suggestion actually was considered!

File: config/admins.txt [hunk 1]
```diff
@@ -139,3 +139,4 @@ Denton-30 = Game Master
 Naksuasdf = Game Master
 MrDoomBringer = Game Master
 shizcalev = Game Master
+NicBR = Game Master
```
Comment (coiax) [inline config/admins.txt]: When I saw that there was the Config Option tag, I was about to shriek that we didn't need being able to click on people under plastic flaps as a config option.

But this is fine.

Comment (subject217) [reply config/admins.txt]: lol

Comment (ATH1909) [reply config/admins.txt]: nice

File: code/controllers/subsystem/vis_overlays.dm [hunk 2]
```diff
@@ -31,8 +31,8 @@ SUBSYSTEM_DEF(vis_overlays)
 			return
 
 //the "thing" var can be anything with vis_contents which includes images
-/datum/controller/subsystem/vis_overlays/proc/add_vis_overlay(atom/movable/thing, icon, iconstate, layer, plane, dir, alpha=255)
-	. = "[icon]|[iconstate]|[layer]|[plane]|[dir]|[alpha]"
+/datum/controller/subsystem/vis_overlays/proc/add_vis_overlay(atom/movable/thing, icon, iconstate, layer, plane, dir, alpha = 255, add_appearance_flags = 0)
```
Comment (ExcessiveUseOfCobblestone) [inline code/controllers/subsystem/vis_overlays.dm]:
```suggestion
/datum/controller/subsystem/vis_overlays/proc/add_vis_overlay(atom/movable/thing, icon, iconstate, layer, plane, dir, alpha = 255, add_appearance_flags = NONE)
```

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]

**Review by ExcessiveUseOfCobblestone — APPROVED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
why did I self-assign this and never do anything with it hmm....

**Pull request merged**